tests/unit/store/createStore.ts

```diff
@@ -429,12 +429,12 @@ registerSuite({
 
 	'async storage': {
 		'async operation should not be done immediately.'(this: any) {
```

Comment:
The memory implementation of the `Storage` interface always returns asynchronously, because it wraps the result in a promise, but the changes are actually executed synchronously since they're just operations on an array in memory. This async `Storage` implementation actually executes the underlying operations asynchronously. 

We found a few bugs where we were implicitly relying on the operations on the underlying storage to be executed in order. That was the original motivation for adding these tests.